pontoon/base/views.py

```diff
@@ -436,39 +435,62 @@ def get_translation_history(request):
     return JsonResponse(payload, safe=False)
 
 
+@utils.require_AJAX
+def get_team_comments(request):
+    """Get team comments for given locale."""
+    try:
+        entity = request.GET["entity"]
+        locale = request.GET["locale"]
+    except MultiValueDictKeyError as e:
+        return JsonResponse(
+            {"status": False, "message": "Bad Request: {error}".format(error=e)},
+            status=400,
+        )
+
+    entity = get_object_or_404(Entity, pk=entity)
+    locale = get_object_or_404(Locale, code=locale)
+    comments = Comment.objects.filter(entity=entity, locale=locale)
+
+    payload = [c.serialize() for c in comments]
+
+    return JsonResponse(payload, safe=False)
+
+
 @require_POST
 @utils.require_AJAX
 @login_required(redirect_field_name="", login_url="/403")
 @transaction.atomic
 def add_comment(request):
-    # TODO: Remove as part of bug 1361318
-    return JsonResponse({"status": False, "message": "Not Implemented"}, status=501,)
-
     """Add a comment."""
-    try:
-        comment = request.POST["comment"]
-        comment = bleach.clean(
-            comment,
-            strip=True,
-            tags=settings.ALLOWED_TAGS,
-            attributes=settings.ALLOWED_ATTRIBUTES,
-        )
-        translationId = request.POST["translationId"]
-    except MultiValueDictKeyError as e:
+    form = forms.AddCommentsForm(request.POST)
+    if not form.is_valid():
         return JsonResponse(
-            {"status": False, "message": "Bad Request: {error}".format(error=e)},
+            {
+                "status": False,
+                "message": "{error}".format(
+                    error=form.errors.as_json(escape_html=True)
+                ),
+            },
             status=400,
         )
 
     user = request.user
-    translation = get_object_or_404(Translation, pk=translationId)
+    comment = form.cleaned_data["comment"]
+    translationId = form.cleaned_data["translation"]
+    entity = get_object_or_404(Entity, pk=form.cleaned_data["entity"])
+    locale = get_object_or_404(Locale, code=form.cleaned_data["locale"])
+    if translationId:
+        translation = get_object_or_404(Translation, pk=translationId)
 
-    c = Comment(author=user, translation=translation, content=comment,)
+    if translationId:
+        c = Comment(author=user, translation=translation, content=comment)
+        log_action("translation_comment:added", user, translation=translation)
+    else:
+        c = Comment(author=user, entity=entity, locale=locale, content=comment)
+        log_action("team_comment:added", user, entity=entity, locale=locale)
```

Comment:
I don't think we need to log two separate actions - we can see what comment type it is by looking at the arguments (translation vs. entity+locale). So I'd revert this change (and related model, validation and migration changes).

Reply:
That is how I originally set this up, but had issues because what is True for one comment is False for the other. So, what validation error do I apply?

Reply:
The logic should be the same as in the `Comment` model:
https://github.com/mozilla/pontoon/pull/1561/files#diff-c18ce63ddb8aca155c0553a0c1b174e3R3449